rclpy/rclpy/clock.py

```diff
@@ -84,15 +83,13 @@ def __init__(self, *, clock, threshold, pre_callback, post_callback):
         if threshold.min_backward is not None:
             min_backward = threshold.min_backward.nanoseconds
 
-        with self._clock.handle as clock_capsule:
-            _rclpy.rclpy_add_clock_callback(
-                clock_capsule, self, threshold.on_clock_change, min_forward, min_backward)
+        self._clock.handle.add_clock_callback(
```

Comment:
Please keep the `with` statement so that the clock can't be destroyed while this is called.

```python3
with self._clock.handle:
    self._clock.handle.add_clock_callback
```